src/fe-write.cc

```diff
@@ -62,6 +62,8 @@ int mainWrite(int argc, const char* argv[])
     flags.parseFlagsWithConfigFiles(argc, argv, formats);
 
 	std::unique_ptr<ImageReader> reader(ImageReader::create(config.image_reader()));
+	reader->preloadImage();
```

Comment:
Is there any reason you can't just load the Image here and then  pass that to writeDiskCommand rather than an ImageReader?

Reply:
That's a very good point --- passing the image to writeDiskCommand makes a lot more sense as the internal interface works with Image anyway, and this gets rid of some reader state.